### PR TITLE
Use ReplaceNodeContent in persist for transformations

### PR DESCRIPTION
**Pull request by gordon-klotho**
Utilize the changes in https://github.com/klothoplatform/klotho/pull/20 to simplify persist plugin.

### Standard checks

- **Unit tests**: Updated unit tests
- **Docs**: internal only
- **Backwards compatibility**: No breaking changes

### CODE REVIEW

File: pkg/lang/javascript/plugin_persist.go [hunk 1]
```diff
@@ -153,99 +153,81 @@ func (p *persister) handleFiles(unit *core.ExecutionUnit) error {
...
+		keyType, pResult := p.determinePersistType(f, annot)
+
 		if len(cap.ID) == 0 {
 			errs.Append(core.NewCompilerError(f, annot, errors.New("'id' is required")))
 		}
 
-		var doTransform func(original *core.SourceFile, modified *core.SourceFile, cap *core.Annotation, result *persistResult, unit *core.ExecutionUnit) (core.CloudResource, error)
-		var err error
+		var resource core.CloudResource
+		var err, runtimeErr, transformErr error
```
Comment (yuval-klotho) [inline pkg/lang/javascript/plugin_persist.go:177]: minor, but what if we instead had this be a `multierr.Error`, and then always appended each err to it (which I _think_ is a noop if the err you're appending is nil?), and then at line 204-210 just checked if the multierr is non-empty? I think that would simplify things, at least to me; fewer vars to think about .

Definitely non-blocking.

**Review by yuval-klotho — APPROVED** [review]
Nice — I like how this reduces the boilerplate!

File: pkg/lang/javascript/plugin_persist.go (hunk 1 above)
Comment (gordon-klotho) [reply pkg/lang/javascript/plugin_persist.go:177]: It's a good idea, but when I tried it I realized the switch block relies on setting the `resource` var (`=` not `:=` so it doesn't go out of scope) so I'd have to define these errors anyway.

Comment (yuval-klotho) [reply pkg/lang/javascript/plugin_persist.go:177]: Never change, Go, never change. 🙃

**Pull request merged**